Insite_AngularUI/src/app/_services/data.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from "@angular/core";
+import { HttpClient } from "@angular/common/http";
+import { BehaviorSubject } from "rxjs";
+import { map } from "rxjs/operators";
+
+import { Attendance } from "../_models/attendance";
+
+@Injectable({ providedIn: "root" })
+export class DataService {
+  constructor(private http: HttpClient) {}
+
+  attendanceList: BehaviorSubject<Attendance[]> = new BehaviorSubject<
+    Attendance[]
+  >([]);
+
+  getAttendanceList(): Promise<any> {
+    return new Promise((resolve, reject) => {
+      this.http
+        .get("https://localhost:44376/api/values", {
+          headers: { "X-Requested-With": "XMLHttpRequest" },
+        })
+        .subscribe(
+          (data: Attendance[]) => this.attendanceList.next(data),
+          (error) => console.log("Error: ", error)
+        );
+    });
+  }
```

Comment:
The use of promises to wait for an observable to finish is essentially using two async patterns where only one is required. The [accepted best practice](https://angular.io/guide/http#requesting-data-from-a-server) is to return the observable from your service, and then subscribe to that in your component:
```
// In this service
getAttendance(): Observable<Attendance[]> {
    return this.http.get(url, headers);
}

// In the attendance component:
ngOnInit() {
    this.service.getAttendance().subscribe(a => this.attendances = a);
```

If you want to perform error handling in the service rather than in each component you can do that by [using the pipe function along with catchError](https://angular.io/guide/http#getting-error-details).

I should also mention that URLs should never be hard-coded, which you can probably see above where I had to change the URL for my project to point at a different localhost port.